### PR TITLE
Added some front end logic to fix state errors with new accounts

### PR DESCRIPTION
**Pull request by greatbacon**
This PR fixes an issue where new accounts were bugging out due to missing communities & spaces. This adds some more front end validation to help render the screen for when accounts have no community, or when a community has no spaces.

I also tweaked the styling to make the community nav stand out, & put the chat input box at the bottom of it's component.

### CODE REVIEW

File: src/lib/Workspace.svelte [hunk 1]
```diff
@@ -8,31 +8,40 @@
 
 	export let friends: Member[];
 	export let communities: Community[];
-	let selectedCommunity = communities[0];
+	let selectedCommunity = communities ? communities[0] : {};
```
Comment (ryanatkn) [inline src/lib/Workspace.svelte:11]: should this be:

`let selectedCommunity = communities ? communities[0] : {};`

?

File: src/lib/Workspace.svelte [hunk 2]
```diff
@@ -8,31 +8,40 @@
...
+	let selectedCommunitySpaces = selectedCommunity ? selectedCommunity.spaces : [];
```
Comment (ryanatkn) [inline src/lib/Workspace.svelte:12]: `let selectedCommunitySpaces = selectedCommunity ? selectedCommunity.spaces : null;`

I think

File: src/lib/Workspace.svelte [hunk 3]
```diff
@@ -8,31 +8,40 @@
...
 	const selectCommunity = (community: Community) => {
 		selectedCommunity = community;
+		selectedCommunitySpaces = community.spaces;
```
Comment (ryanatkn) [inline src/lib/Workspace.svelte:15]: remove, can be reactive

File: src/lib/Workspace.svelte [hunk 4]
```diff
@@ -8,31 +8,40 @@
...
 	};
-	let selectedSpace = selectedCommunity.spaces[0];
+	let selectedSpace = selectedCommunity ? selectedCommunity.spaces[0] : null;
 	const selectSpace = (space: Space) => {
 		selectedSpace = space;
 		console.log(`[ss] ${selectedSpace.url}`);
 	};
+	if (selectedSpace) {
+		console.log('selectedSpace', selectedSpace);
+	}
```
Comment (ryanatkn) [inline src/lib/Workspace.svelte:24]: could change these three lines to:

`$: console.log('selectedSpace', selectedSpace)`

to get reactive logging

**Review by ryanatkn — APPROVED** [review]

File: src/lib/Workspace.svelte (hunk 1 above)
Comment (greatbacon) [reply src/lib/Workspace.svelte:11]: I wasn't sure what was better idiomatically so I went with empty object. If we're in the clear to use `null` here, I'll happily use null here lol

Comment (ryanatkn) [reply src/lib/Workspace.svelte:11]: Yeah gotcha, in this case prefer `null`, empty object is almost never something we'd do in JS -- empty arrays are common, but in this case we want to use `null` meaningfully.

**Pull request merged**